server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -387,14 +385,15 @@ private void updateMemoryInfo(
     tsFileProcessorInfo.addTSPMemCost(unsealedResourceIncrement + chunkMetadataIncrement);
     if (storageGroupInfo.needToReportToSystem()) {
       SystemInfo.getInstance().reportStorageGroupStatus(storageGroupInfo);
-      try {
-        StorageEngine.blockInsertionIfReject();
-      } catch (WriteProcessRejectException e) {
-        storageGroupInfo.releaseStorageGroupMemCost(memTableIncrement);
-        tsFileProcessorInfo.releaseTSPMemCost(unsealedResourceIncrement + chunkMetadataIncrement);
-        SystemInfo.getInstance().resetStorageGroupStatus(storageGroupInfo, false);
-        throw e;
+      long startTime = System.currentTimeMillis();
+      while (SystemInfo.getInstance().isRejected()) {
+        try {
+          storageGroupInfo.getStorageGroupProcessor().rejectConditionAwait();
```

Comment:
So, if we wait to timeout, we could not return now?

Reply:
Yes, I remove the In-WriteLock timeout exception. 
There are still the out-writeLock timeout exception in there. How do you think?

Reply:
Ok, I add the timeout back....